src/main/java/ro/hiringsystem/service/impl/ManagerUserServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package ro.hiringsystem.service.impl;
+
+import ro.hiringsystem.model.ManagerUser;
+import ro.hiringsystem.service.ManagerUserService;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class ManagerUserServiceImpl implements ManagerUserService {
```

Comment:
Since some methods of the implementaions are also very similar, the implementation itself can also be made a template and extended if custom methods are needed. (just as for the interfaces)

Reply:
I'm afraid this wouldn't work as expected, because it would prevent us from having four different static maps for all types of users. Hence, I leave the implementations without a template.